### PR TITLE
Refactor oauth2 client state and prefill it with initial values

### PR DESCRIPTION
**Pull request by Tug**
New attempt for https://github.com/Automattic/wp-calypso/pull/17166

----

This PR is a refactoring and moves the oauth2 clients data from the new oauth2 layout to the `login.oauth2.clients` state.

### Testing Instructions
- Boot this branch
- Navigate to http://calypso.localhost:3000/log-in?client_id=930 while logged out
- check that the Master Bar is removed
- check that VaultPress is mentioned: `Howdy! Log in to VaultPress with your WordPress.com account.` and `Back to VaultPress` link at the bottom.
- check that `/log-in` still loads as expected for logged in users

### Reviews
- [x] Code
- [x] Product
- [x] Tests

### CODE REVIEW

**Comment by Tug** [comment]
The actual fix for the bug is in https://github.com/Automattic/wp-calypso/pull/17187/commits/e64e90c6c8a56ca4b9a16882471411d327e25773

**Comment by pento** [comment]
👍🏻

**Comment by Tug** [comment]
> Why do we allow authenticated users to view /log-in? Wouldn't it make more sense to redirect those users to /?

It's actually a feature. I think the reason behind it is that we want users to be able to switch accounts easily.

File: client/state/login/actions.js [hunk 1]
```diff
@@ -107,7 +104,7 @@ function getErrorFromHTTPError( httpError ) {
  * @param {Object} wpcomError HTTP error
  * @returns {{message: string, field: string, code: string}} an error message and the id of the corresponding field
  */
-const getErrorFromWPCOMError = ( wpcomError ) => ( {
+export const getErrorFromWPCOMError = ( wpcomError ) => ( {
```
Comment (Tug) [inline client/state/login/actions.js]: Nop, and I thought I did remove it, fixing

File: client/layout/oauth-client.jsx [hunk 2]
```diff
@@ -10,58 +10,25 @@ import { connect } from 'react-redux';
...
 
 	const classes = classNames( 'layout', {
 		[ 'is-group-' + section.group ]: !! section,
 		[ 'is-section-' + section.name ]: !! section,
 		'focus-content': true,
 		'has-no-sidebar': true, // Logged-out never has a sidebar
 		'wp-singletree-layout': !! primary,
-		dops: !! client,
-		[ client.name ]: !! client,
+		dops: hasValidOAuth2ClientData,
```
Comment (Tug) [inline client/layout/oauth-client.jsx]: That could make sense I agree. However this PR is not related to the actual customization of those pages so I'd say let's follow the YAGNI principle for now.
